lib/lz4.c

```diff
@@ -188,7 +188,14 @@
 /*-************************************
 *  Memory routines
 **************************************/
-#ifdef LZ4_USER_MEMORY_FUNCTIONS
+#if defined(LZ4_STATIC_LINKING_ONLY_DISABLE_MEMORY_ALLOCATION)
+#  undef LZ4_ALLOC
+#  undef LZ4_ALLOC_AND_ZERO
+#  undef LZ4_FREEMEM
+#  define LZ4_ALLOC(x)			lz4_error_memory_allocation_is_disabled
```

Comment:
If I understand properly, these macros `LZ4_ALLOC(x)` and next are designed to fail at compile time, as a way to detect if they are employed anywhere.

However, these macros are also not used anywhere.

Did you mean to replace `ALLOC(s)` ?

Reply:
oops.  You're right.  They must be `ALLOC(s)`, `ALLOC_AND_ZERO(s)` and `FREEMEM(p)`.